app/views/interactions/index.html.erb

```diff
@@ -29,12 +29,12 @@
           </td>
           <td class="name">
             <%= interaction.label %>
-            <p><%= display_link_for(interaction) %></p>
+            <p><%= link = display_link_for(interaction) %></p>
           </td>
           <td>
           </td>
           <td class='text-center'>
-            <%= link_to('Edit link', edit_local_authority_service_interaction_links_path(interaction_slug: interaction.slug), class: "btn btn-default btn-s") %>
+            <%= link_to(link == 'No link' ? 'Add link' : 'Edit link', edit_local_authority_service_interaction_links_path(interaction_slug: interaction.slug), class: "btn btn-default btn-s") %>
```

Comment:
`display_link_for` doesn't talk to the database.  The controller fetches everything from the db and puts it in a hash (that's what `@links` is), and `display_link_for` looks up entries in that hash.